ciao-cli/configuration_test.go

```diff
@@ -0,0 +1,173 @@
+//
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+package main
+
+import "testing"
+
+var configURICases = []struct {
+	uri      string // URI to evaluate
+	scheme   string // expected case to handle validation
+	expected bool   // expected result of the test case
+}{
+	{"http://identity.example.com:35357", "http", false},   // invalid URI (insecure scheme)
+	{"http://identity.example.com:35357", "https", false},  // invalid URI (invalid scheme)
+	{"https://:8080/subdir", "https", false},               // invalid URI (no host)
+	{"file://no-path-defined.example.com", "file", false},  // invalid URI (no path)
+	{"https://some-host.example.com:35357", "https", true}, // valid URI (host defined)
+	{"file:///etc/ciao/configuration.yaml", "file", true},  // valid URI (path defined)
+}
+
+var configBooleanCases = []struct {
+	boolStr  string // boolean string to evaluate
+	saneBool string // boolean value sanatized
+	expected bool   // expected result of the test case
+}{
+	{"1", "true", true},
+	{"t", "true", true},
+	{"T", "true", true},
+	{"true", "true", true},
+	{"TruE", "true", true},
+	{"TRUE", "true", true},
+	{"0", "false", true},
+	{"f", "false", true},
+	{"F", "false", true},
+	{"false", "false", true},
+	{"FalsE", "false", true},
+	{"FALSE", "false", true},
+	{"invalid", "", false},
+}
```

Comment:
I'm happy to re-work on this to make it more generic if needed in a later PR.

For the string input validator, using the [Set(val string) error](https://github.com/01org/ciao/pull/1130#discussion_r103516816) method is useful, but for the second flag I wasn't able to use it because of the dependence in other flag to determine it's data type.